assets/js/modules/tagmanager/datastore/containers.js

```diff
@@ -0,0 +1,171 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { STORE_NAME, CONTEXT_WEB, CONTEXT_AMP } from './constants';
+import { isValidAccountID, isValidUsageContext } from '../util/validation';
+import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
+
+const fetchGetContainersStore = createFetchStore( {
+	baseName: 'getContainers',
+	argsToParams: ( accountID ) => {
+		invariant( isValidAccountID( accountID ), 'A valid accountID is required to fetch containers.' );
+
+		return { accountID };
+	},
+	controlCallback: ( { accountID } ) => {
+		// Always request both contexts to prevent filtering on server.
+		// TODO: Remove `usageContext` param when legacy component is removed and datapoint
+		// defaults to returning all containers if no context is provided.
+		const usageContext = [ CONTEXT_WEB, CONTEXT_AMP ];
+		return API.get( 'modules', 'tagmanager', 'containers', { accountID, usageContext }, { useCache: false } );
+	},
+	reducerCallback: ( state, containers, { accountID } ) => {
+		return {
+			...state,
+			containers: {
+				...state.containers,
+				[ accountID ]: [ ...containers ],
+			},
+		};
+	},
+} );
+
+const fetchCreateContainerStore = createFetchStore( {
+	baseName: 'createContainer',
+	argsToParams( accountID, usageContext ) {
+		invariant( isValidAccountID( accountID ), 'A valid accountID is required to create a container.' );
+		invariant( isValidUsageContext( usageContext ), 'A valid usageContext is required to create a container.' );
+
+		return { accountID, usageContext };
+	},
+	controlCallback: ( { accountID, usageContext } ) => {
+		return API.set( 'modules', 'tagmanager', 'create-container', { accountID, usageContext } );
+	},
+	reducerCallback( state, container, { accountID } ) {
+		return {
+			...state,
+			containers: {
+				...state.containers,
+				[ accountID ]: [
+					...( state.containers[ accountID ] || [] ),
+					container,
+				],
+			},
+		};
+	},
+} );
+
+const BASE_INITIAL_STATE = {
+	containers: {},
+};
+
+const baseActions = {
+	/**
+	 * Creates a new Tag Manager container in the given account.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} accountID    Google Tag Manager account ID.
+	 * @param {string} usageContext Container usage context. (Either 'web', or 'amp')
+	 * @return {Object} Object with `response` and `error`.
+	 */
+	*createContainer( accountID, usageContext ) {
```

Comment:
Along the lines of what we decided in #716, I wonder whether we should make `usageContext` property of an object? We could keep it required or alternatively default to `web`. This diverges from the ACs, but wanted to put it up for consideration.

Reply:
I like this idea although we should be careful about using defaults in the datastore due to the high possibility of `undefined` being passed for something. It's not unreasonable to think that we might allow for the container name to be customized at creation time - it's actually more important than a profile name since you can't have two containers with the same name, but there are almost no restrictions on view names.

Reply:
Sounds good, let's keep it required for now, but move into an object since it could easily become one of _multiple_ creation arguments.